src/Rector/FuncCall/ArgumentFuncCallToMethodCallRector.php

```diff
@@ -0,0 +1,254 @@
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\Exception\ShouldNotHappenException;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Naming\Naming\PropertyNaming;
+use Rector\Naming\ValueObject\ExpectedName;
+use Rector\NodeTypeResolver\TypeAnalyzer\ArrayTypeAnalyzer;
+use Rector\PostRector\Collector\PropertyToAddCollector;
+use Rector\PostRector\ValueObject\PropertyMetadata;
+use Rector\StaticTypeMapper\ValueObject\Type\FullyQualifiedObjectType;
+use RectorLaravel\Contract\ValueObject\ArgumentFuncCallToMethodCallInterface;
+use RectorLaravel\ValueObject\ArgumentFuncCallToMethodCall;
+use RectorLaravel\ValueObject\ArrayFuncCallToMethodCall;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+
+/**
+ * @see \RectorLaravel\Tests\Rector\FuncCall\ArgumentFuncCallToMethodCallRector\ArgumentFuncCallToMethodCallRectorTest
+ */
+final class ArgumentFuncCallToMethodCallRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    /**
+     * @var ArgumentFuncCallToMethodCallInterface[]
+     */
+    private array $argumentFuncCallToMethodCalls = [];
+
+    public function __construct(
+        private readonly ArrayTypeAnalyzer $arrayTypeAnalyzer,
+        private readonly PropertyNaming $propertyNaming,
+        private readonly PropertyToAddCollector $propertyToAddCollector
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Move help facade-like function calls to constructor injection', [
+            new ConfiguredCodeSample(
+                <<<'CODE_SAMPLE'
+class SomeController
+{
+    public function action()
+    {
+        $template = view('template.blade');
+        $viewFactory = view();
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+class SomeController
+{
+    /**
+     * @var \Illuminate\Contracts\View\Factory
+     */
+    private $viewFactory;
+
+    public function __construct(\Illuminate\Contracts\View\Factory $viewFactory)
+    {
+        $this->viewFactory = $viewFactory;
+    }
+
+    public function action()
+    {
+        $template = $this->viewFactory->make('template.blade');
+        $viewFactory = $this->viewFactory;
+    }
+}
+CODE_SAMPLE
+                ,
+                [new ArgumentFuncCallToMethodCall('view', 'Illuminate\Contracts\View\Factory', 'make')]
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
+    }
+
+    /**
+     * @param FuncCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkipFuncCall($node)) {
+            return null;
+        }
+
+        /** @var Class_ $classLike */
+        $classLike = $this->betterNodeFinder->findParentType($node, Class_::class);
```

Comment:
instead of find parent type, it is better to use `Class_` node, and traverser its methods, something like this:

```php
        $hasChanged = false;
        $this->traverseNodesWithCallable($node->getMethods(), function (Node $node) use (
            &$hasChanged
        ) {
                if ($node instanceof  FuncCall) {
                       $result = $this->refactorFuncCallToMethodCall($node);
                       if ($result instanceof MethodCall) {
                              $hasChanged = true;
                              return $result;
                       }

                       return null;
                }

                return null;
        });

        if ($hasChanged) {
              return $node;
        }

        return null;
```